server/graphql/v1/mutations/orders.js

```diff
@@ -574,6 +611,20 @@ export async function createOrder(order, req) {
         }
 
         await orderCreated.setPaymentMethod(order.paymentMethod);
+
+        // Update all existing orders with the new payment method
+        if (get(order, 'paymentMethod.service') === 'stripe' && order.paymentMethod.data?.fingerprint) {
+          // Check if the payment method is already saved
+          await sequelize.transaction(async transaction => {
+            const oldPaymentMethod = await deletePaymentMethod(
+              fromCollective,
+              order.paymentMethod.data.fingerprint,
+              transaction,
+            );
+
+            await updateExistingOrdersToNewPaymentMethod(orderCreated.paymentMethod, oldPaymentMethod, transaction);
```

Comment:
:ice_cream: I would find it clearer to have the condition here. You're currently forced to go down into `deletePaymentMethod` + `updateExistingOrdersToNewPaymentMethod` to understand what's happening here.

```suggestion
            if (oldPaymentMethod) {
              await updateExistingOrdersToNewPaymentMethod(orderCreated.paymentMethod, oldPaymentMethod, transaction);
            }
```